metrics/lib/common.bash

```diff
@@ -95,18 +95,42 @@ check_images()
 # This function performs a docker build on the image names
 # passed in, to ensure that we have the latest changes from
 # the dockerfiles
+build_dockerfile_image()
+{
+	local image="$1"
+	local dockerfile_path="$2"
+
+	echo "docker building $image"
+	if ! docker build --label "$image" --tag "${image}:latest" -< "$dockerfile_path"; then
+		die "Failed to docker build image $image"
+		exit 1;
+	fi
+}
+
+# This function verifies that the dockerfile version is
+# equal to the test version in order to build the image or
+# just run the test
 check_dockerfiles_images()
 {
 	local image="$1"
 	local dockerfile_path="$2"
+	local test_version="$3"
+
 	if [ -z "$image" ] || [ -z "$dockerfile_path" ]; then
 		die "Missing image or dockerfile path variable"
 		exit 1;
 	fi
-	echo "docker building $image"
-	if ! docker build --label "$image" --tag "${image}:latest" -< "$dockerfile_path"; then
-		die "Failed to docker build image $image"
-		exit 1;
+
+	# Verify that dockerfile version is equal to test version
+	check_image=$(docker images "$image")
+	if [ -z "$check_image" ]; then
+		dockerfile_version=$(cat "$dockerfile_path" | grep "DOCKERFILE_VERSION" | cut -d '=' -f2)
```

Comment:
I was thinking maybe more what we'd do would use a
```
LABEL Version="1.3"
```
in the Dockerfile (that then shows up in `docker inspect` easily I believe, and then just check the `docker inspect image` against the grep of the `Version` from the Dockerfile source. I don't think we need to have TEST_VERSION in the script itself - we are (I think) only interested if the contents of the `Dockerfile` change, and thus we need an image rebuild.

Oh, and 'I c a r u s' would note that you don't need the `cat | grep` - you can just put the filename on the end of the grep command (`grep Version filename`), which saves us spawning an extra process for the `cat` ;-)  (I blame 'I c a r u s' ;-) )